### PR TITLE
plugins/wrapping: Add more options

### PR DESCRIPTION
**Pull request by ZainKergaye**
Adding more options to wrapping.nvim. Please let me know if anything is amiss. Thank you!

### CODE REVIEW

File: plugins/by-name/wrapping/default.nix [hunk 1]
```diff
@@ -18,9 +14,60 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
...
+      defaultNullOpts.mkListOf lib.types.str [ ]
+        "Auto set mode filetype deny list";
+    auto_set_mode_heuristically = defaultNullOpts.mkBool true ''
+       By default, the plugin will set the hard or soft
+      mode automatically when any file loads (for a
+      specific set of file types, see below), using the
+      BufWinEnter event in an autocmd. It uses a variety
+      of heuristics (which aren't documented in detail here
+      as they will evolve over time as wrapping.nvim becomes
+      more sophisticated).'';
```
Comment (khaneliman) [inline plugins/by-name/wrapping/default.nix:57]:
```suggestion
      If true, the plugin will set the hard or soft mode automatically when any file loads. 
      
      For a specific set of file types, use `plugins.wrapping.settings.auto_set_mode_filetype_allowlist`.
      
      It uses the `BufWinEnter` event in an autocmd with a variety of undocumented heuristics.
      Refer to the [plugin documentation] for more details on this evolving behavior.
      [plugin documenation] https://github.com/andrewferrier/wrapping.nvim?tab=readme-ov-file#automatic-heuristic-mode'';
      
```

File: plugins/by-name/wrapping/default.nix [hunk 2]
```diff
@@ -18,9 +14,60 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
...
+    set_nvim_opt_default = defaultNullOpts.mkBool true ''
+      By default, wrapping.nvim will tweak some NeoVim defaults
```
Comment (khaneliman) [inline plugins/by-name/wrapping/default.nix:59]:
```suggestion
      If true, wrapping.nvim will tweak some NeoVim defaults
```

**Review by khaneliman — COMMENTED** [review]
Just some suggestions for rewording their documentation to better fit in our docs and a default value fix.

**Comment by ZainKergaye** [comment]
Thanks! Just comitted those, should I squash those commits? I need some guidance if so

**Comment by khaneliman** [comment]
Looks like formatting check is failing:

       > -      
       > +
       >        It uses the `BufWinEnter` event in an autocmd with a variety of undocumented heuristics.
       >        Refer to the [plugin documentation] for more details on this evolving behavior.
       >        [plugin documenation] https://github.com/andrewferrier/wrapping.nvim?tab=readme-ov-file#automatic-heuristic-mode'';
       > -      
       > +
       >      set_nvim_opt_default = defaultNullOpts.mkBool true ''
       >        If true, wrapping.nvim will tweak some NeoVim defaults
       >        (linebreak and wrap) to make it operate more smoothly.

**Comment by khaneliman** [comment]
Pushed the cleanup, you can rebase it on latest from main if you want to get yourself back to the only author.

File: plugins/by-name/wrapping/default.nix [hunk 3]
```diff
@@ -18,9 +14,68 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
       By default, wrapping.nvim will output a message to the
       command line when the hard or soft mode is set.
     '';
+
+    create_commands = defaultNullOpts.mkBool true ''
+      If true, the plugin will create the following
+      commands to set/override a wrapping mode in case
+      it is not autodetected correctly:
+
+      * HardWrapMode
```
Comment (MattSturgeon) [inline plugins/by-name/wrapping/default.nix:23]:
```suggestion
      * `HardWrapMode`
```

Where we have symbol names, mappings, or code snippets it's best to use `` `backticks` `` to render a `<code>` span.

File: plugins/by-name/wrapping/default.nix [hunk 4]
```diff
@@ -18,9 +14,68 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
...
+
+      * SoftWrapMode
+
+      * ToggleWrapMode
+    '';
+
+    create_keymaps = defaultNullOpts.mkBool true ''
+      If true, it will create the following normal-mode keymappings:
+
+      * [ow (soft wrap mode)
```
Comment (MattSturgeon) [inline plugins/by-name/wrapping/default.nix:33]:
```suggestion
      * `[ow` (soft wrap mode)
```

Another example :slightly_smiling_face:

File: plugins/by-name/wrapping/default.nix [hunk 5]
```diff
@@ -18,9 +14,68 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
...
+        "Auto set mode filetype deny list";
+
+    auto_set_mode_heuristically = defaultNullOpts.mkBool true ''
+      If true, the plugin will set the hard or soft mode automatically when any file loads.
+
+      For a specific set of file types, use `plugins.wrapping.settings.auto_set_mode_filetype_allowlist`.
+
+      It uses the `BufWinEnter` event in an autocmd with a variety of undocumented heuristics.
+      Refer to the [plugin documentation] for more details on this evolving behavior.
+      [plugin documentation] https://github.com/andrewferrier/wrapping.nvim?tab=readme-ov-file#automatic-heuristic-mode'';
```
Comment (MattSturgeon) [inline plugins/by-name/wrapping/default.nix:64]:
```suggestion
      Refer to the [plugin documentation] for more details on this evolving behavior.

      [plugin documentation] https://github.com/andrewferrier/wrapping.nvim?tab=readme-ov-file#automatic-heuristic-mode'';
```

We should ideally have a blankline before any footnote links.

**Review by MattSturgeon — COMMENTED** [review]
Looks good, I would suggest using \`backticks\` more frequently, e.g. for key-mapping examples or when quoting autocmd event names, highlight groups, etc.

**Review by MattSturgeon — APPROVED** [review]
LGTM once squashed/rebased :grin:

**Comment by ZainKergaye** [comment]
Could you direct me on how to rebase this properly? I am using 'git rebase -i 5ed2526' and given an interactive editor to put a bunch of commands into a text file but messing around with 'squash d83c133c plugins/wrappign: Add more options' does nothing but say it can't squash without a previous commit. 

Thanks for the guidance!

**Comment by MattSturgeon** [comment]
> Could you direct me on how to rebase this properly? I am using `git rebase -i 5ed2526`

`git rebase 5ed2526` will rebase _onto_ 5ed2526, however you want 5ed2526 to be _part of_ the commits _being_ rebased.

You could use `^` to get the commit's parent: `git rebase 5ed2526^` or `HEAD^^`, however this won't achieve much since we actually want to rebase onto nixvim's latest `main`.

Instead you can do:
```
git fetch upstream
git rebase upstream/main
```

As for `-i` and/or squashing:

When you use `-i` you get an interactive list of the commits that will be rebased _onto_ the geven target, in the order they will be applied.

Normally you'll want the list to look like:
```
pick 5ed2526
squash d83c133
```

As this is a fairly simple example, you could also leverage `git reset` and `git commit --amend`:
```
git reset --soft HEAD^
git commit --amend --no-edit
```

Once all this is done, you'll have to do a "force" push, since your local branch will have differing history to the remote branch:
```
git push --force-with-lease
```

**Comment by ZainKergaye** [comment]
Got it! Thanks! I would love to learn more about rebasing but your simple 'git reset --soft HEAD^' and 'push' worked like a charm. Thanks!

**Comment by MattSturgeon** [comment]
Awesome!

Note that you've squashed (manually, using `reset` & `commit`), but you haven't rebased onto our `main` branch.
image

To do this you should:
```
git fetch upstream
git rebase upstream/main
git push --force-with-lease
```

**Pull request merged**